mongox/models.py

```diff
@@ -114,6 +114,25 @@ async def get(self) -> T:
             raise MultipleMatchesFound()
         return objects[0]
 
+    async def get_or_create(self, defaults: typing.Dict = dict()) -> T:
+        """
+        Get the only document matching or created the document
+        """
+
+        objects = await self.limit(2).all()
+        if len(objects) == 2:
+            raise MultipleMatchesFound()
```

Comment:
I think we can make use if the `get` method which is already handling these queries.

And for the create too. I suspect this should use get and create from existing ones.

Reply:
I do't use the `get` method because it generates an exception if there is no document that makes matched and controlling the flow with our own exceptions does't seem to me to be the best way to do it.

Reply:
Yes maybe something like this:

```python
try:
    get()
except NoMatchFound:
    create()
```

And if MultipleMatchFound we will raise the exception.

Reply:
ok, today I will take some time to organize it this way.

Reply:
Have you had any progress with this? I'm really interested to get this through.

Reply:
yes, the truth is that a few days ago I uploaded it. The last commits are just for this.

Reply:
Sorry my comment wasn't very clear. I'll explain further down.